internal/monitor_conf.go

```diff
@@ -104,70 +190,124 @@ func getStartCommandLines(cluster *fdbtypes.FoundationDBCluster, processClass fd
+			{ArgumentType: monitorapi.EnvironmentArgumentType, Source: "FDB_MACHINE_ID"},
+		}},
+		monitorapi.Argument{ArgumentType: monitorapi.ConcatenateArgumentType, Values: []monitorapi.Argument{
+			{Value: "--locality_zoneid="},
+			{ArgumentType: monitorapi.EnvironmentArgumentType, Source: zoneVariable},
+		}},
+	)
+
+	if cluster.NeedsExplicitListenAddress() && cluster.Status.HasListenIPsForAllPods {
+		configuration.Arguments = append(configuration.Arguments, monitorapi.Argument{ArgumentType: monitorapi.ConcatenateArgumentType, Values: buildIPArgument("listen_address", "FDB_POD_IP", imageType, sampleAddresses)})
 	}
 
-	confLines = append(confLines,
-		"locality_instance_id = $FDB_INSTANCE_ID",
-		"locality_machineid = $FDB_MACHINE_ID",
-		fmt.Sprintf("locality_zoneid = %s", zoneVariable))
+	if cluster.Spec.MainContainer.PeerVerificationRules != "" {
+		configuration.Arguments = append(configuration.Arguments, monitorapi.Argument{Value: fmt.Sprintf("--tls_verify_peers=%s", cluster.Spec.MainContainer.PeerVerificationRules)})
+	}
+
+	podSettings := cluster.GetProcessSettings(processClass)
+
+	if podSettings.CustomParameters != nil {
+		equalPattern, err := regexp.Compile(`\s*=\s*`)
+		if err != nil {
+			return configuration, err
+		}
+		for _, argument := range *podSettings.CustomParameters {
+			sanitizedArgument := "--" + equalPattern.ReplaceAllString(argument, "=")
+			configuration.Arguments = append(configuration.Arguments, monitorapi.Argument{Value: sanitizedArgument})
+		}
+	}
 
 	if cluster.Spec.DataCenter != "" {
-		confLines = append(confLines, fmt.Sprintf("locality_dcid = %s", cluster.Spec.DataCenter))
+		configuration.Arguments = append(configuration.Arguments, monitorapi.Argument{Value: fmt.Sprintf("--locality_dcid=%s", cluster.Spec.DataCenter)})
 	}
 
 	if cluster.Spec.DataHall != "" {
-		confLines = append(confLines, fmt.Sprintf("locality_data_hall = %s", cluster.Spec.DataHall))
+		configuration.Arguments = append(configuration.Arguments, monitorapi.Argument{Value: fmt.Sprintf("--locality_data_hall=%s", cluster.Spec.DataHall)})
 	}
 
-	if cluster.Spec.MainContainer.PeerVerificationRules != "" {
-		confLines = append(confLines, fmt.Sprintf("tls_verify_peers = %s", cluster.Spec.MainContainer.PeerVerificationRules))
-	}
+	return configuration, nil
+}
 
-	if cluster.NeedsExplicitListenAddress() && cluster.Status.HasListenIPsForAllPods {
-		confLines = append(confLines, fmt.Sprintf("listen_address = %s", fdbtypes.ProcessAddressesString(cluster.GetFullAddressList("$FDB_POD_IP", false, processNumber), ",")))
+// buildIPArgument builds an argument that takes an IP address from an environment variable
+func buildIPArgument(parameter string, environmentVariable string, imageType FDBImageType, sampleAddresses []fdbtypes.ProcessAddress) []monitorapi.Argument {
+	var leftIPWrap string
+	var rightIPWrap string
+	if imageType == FDBImageTypeUnified {
+		leftIPWrap = "["
+		rightIPWrap = "]"
+	} else {
+		leftIPWrap = ""
+		rightIPWrap = ""
```

Comment:
I'm not totally sure if I understand that, why are we adding `[]` for the unified image but not for the split image? I thought that depends in the IP family we want to use and not the image type? Otherwise this will generate the wrong format for IPv4/IPv6?

Reply:
Could you add some context here (preferable as a comment).